File: 3.从尾到头打印链表.cpp
/**
*  struct ListNode {
*        int val;
*        struct ListNode *next;
*        ListNode(int x) :
*              val(x), next(NULL) {
*        }
*  };
*/
class Solution {
public:
    vector<int> printListFromTailToHead(ListNode* head) {
        stack<int> istk;
        vector<int> ret;
        while(head != nullptr){
            istk.push(head->val);
            head = head -> next;
        }
        while(!istk.empty()){
            ret.push_back(istk.top());
            istk.pop();
        }
        return ret;
    }
};

File: java/3.从尾到头打印链表.java
/**
*    public class ListNode {
*        int val;
*        ListNode next = null;
*
*        ListNode(int val) {
*            this.val = val;
*        }
*    }
*
*/
import java.util.ArrayList;
public class Solution {
    ArrayList<Integer> ret = new ArrayList<Integer>();
    public ArrayList<Integer> printListFromTailToHead(ListNode listNode) {
        dfs(ret, listNode);
        return ret;
    }
    public void dfs(ArrayList<Integer> arr, ListNode pHead) {
        if(pHead == null)
            return;
        dfs(arr, pHead.next);
        ret.add(pHead.val);
    }
}
